### PR TITLE
prepare audio release

### PR DESCRIPTION
**Pull request by 7flash**
add home background asset

add isRecommended and tags fields for models

rename AppStore to AppController

rename StoreService to FirestoreService

add loop mode for audio player

change skip button icon in audio player

add recommended audio on home page

remove unused modules

implement discovery audio screen

implement discovery video screen

implement discovery authors screen

separate users list widget for profile followers/following

modify home page layout

add essential animations - fade in page transition, hero widgets, and animated opacity

DISABLE videos and video player for this release

### CODE REVIEW

File: lib/modules/discover/videos/discover_videos_view.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          return RepeatWidget(controller.loadVideos);
+        default:
+          return SplashWidget();
+      }
+    });
+  }
+
+  Widget buildVideos(List<Video> videos) => ListView.builder(
+        itemCount: videos.length,
+        itemBuilder: (_, index) => VideoCard(videos[index], onPlayTap: () => VideoPlayerModule.toPlayer(videos[index])),
```
Comment (minikin) [inline lib/modules/discover/videos/discover_videos_view.dart:41]: We changed API for presenting VideoPlayer, please [check](https://github.com/HighVibe/hvn/blob/8312139afe0be57ff436cccc7a89abd1450f3753/lib/modules/home/widgets/header_widget.dart#L108).

File: lib/modules/discover/videos/video_card.dart [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            top: 10,
+            right: 10,
+            child: Container(
+              padding: const EdgeInsets.all(8),
+              decoration: BoxDecoration(
+                color: secondaryColor,
+                borderRadius: BorderRadius.circular(50),
+              ),
+              child: Text(
+                Duration(milliseconds: video.fileDetails.durationMs)
```
Comment (minikin) [inline lib/modules/discover/videos/video_card.dart:50]: Can we move string formatting to a free function for maybe create a specific class for it?

File: lib/modules/home/widgets/audios_widget.dart [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          ),
+          SizedBox(
+            height: 240,
+            child: PageView.builder(
+              controller: sliderController,
+              scrollDirection: Axis.horizontal,
+              itemCount: audios.length,
+              itemBuilder: (_, index) => AudioPreviewItem(
+                audio: audios[index],
+                onTap: (item) => AudioPlayerModule.toPlayer(audios[index]),
```
Comment (minikin) [inline lib/modules/home/widgets/audios_widget.dart:64]: We changed API for presenting Audio Player. Please check the previous comment.

File: lib/modules/home/widgets/videos_widget.dart [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+          ),
+          SizedBox(
+            height: 240,
+            child: PageView.builder(
+              controller: sliderController,
+              scrollDirection: Axis.horizontal,
+              itemCount: videos.length,
+              itemBuilder: (_, index) => VideoPreviewItem(
+                video: videos[index],
+                onTap: (item) => VideoPlayerModule.toPlayer(videos[index]),
```
Comment (minikin) [inline lib/modules/home/widgets/videos_widget.dart:64]: We changed API for presenting VideoPlayer. Please check the previous comment.

File: lib/modules/profile/audio/audio_page.dart [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                audioFile: item,
+                onTap: (_) => _showAudioPlayer(context, item),
+              ),
+            )
+            .toList(),
+      ],
+    );
+  }
+
+  void _showAudioPlayer(BuildContext context, Audio audioFile) {
```
Comment (minikin) [inline lib/modules/profile/audio/audio_page.dart:59]: We changed API for presenting AudioPlayer. Please check the previous comment.

File: lib/modules/audio_player/audio_player_controller.dart [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import 'package:flutter_modular/flutter_modular.dart';
+import 'package:highvibe/modules/audio_player/audio_player_service.dart';
+import 'package:mobx/mobx.dart';
+
+part 'audio_player_controller.g.dart';
+
+class AudioPlayerController = _AudioPlayerControllerBase
```
[the remark this answers is not in the document]
Comment (7flash) [reply lib/modules/audio_player/audio_player_controller.dart:7]: Modular structure may increase complexity in short term, but keeps sanity when code base grows over 20k lines, so let's maintain consistency now

File: lib/modules/discover/discover_module.dart [hunk 7]
```diff
@@ -1,29 +1,35 @@
...
 
-  static Future toDiscover({int pageIndex}) => Modular.to.pushNamed(
-        DISCOVER,
-        arguments: pageIndex,
-      );
+  static Future toDiscover({int pageIndex}) =>
+      Modular.to.pushNamed(DISCOVER, arguments: pageIndex);
+
+  static Future toDiscoverAudios() =>
+      Modular.to.pushNamed(DISCOVER, arguments: 0);
```
[the remark this answers is not in the document]
Comment (7flash) [reply lib/modules/discover/discover_module.dart:28]: OK, let's do it. https://github.com/HighVibe/hvn/issues/125

File: lib/modules/discover/videos/discover_videos_view.dart (hunk 1 above)
Comment (7flash) [reply lib/modules/discover/videos/discover_videos_view.dart:41]: I guess, that's what messed up during merge and causes black screen now

File: lib/modules/discover/videos/video_card.dart (hunk 2 above)
Comment (7flash) [reply lib/modules/discover/videos/video_card.dart:50]: Sure, should I move that to utils.dart or another file?

File: lib/modules/audio_player/audio_player_controller.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (minikin) [reply lib/modules/audio_player/audio_player_controller.dart:8]: From my point of view and experience `flutter_modular` can be easily replaced with a couple of generic, value type classes and proper code-style hygiene in a project. Moreover `flutter_modular` is limit us of using even many basic Flutter's SDK features (e.g. custom transition).

[the remark this answers is not in the document]
Comment (7flash) [reply lib/modules/audio_player/audio_player_controller.dart:8]: Let's begin discussing about alternative architecture, but in meantime following this pattern for current features, as for me flutter_modular + mobx feels really good.

[the remark this answers is not in the document]
Comment (7flash) [reply lib/modules/audio_player/audio_player_controller.dart:8]: Custom transitions are supported by flutter_modular with customTransition argument for path

[the remark this answers is not in the document]
Comment (minikin) [reply lib/modules/audio_player/audio_player_controller.dart:8]: @7flash 

My point is that we don't need to use libraries just for the sake of that libraries. `flutter_modular` and `mobx` doesn't add anything for AudioPlayer and VideoPlayer.

-----
With such a way of pushing VideoPlayerPage widget :
```
static Future toPlayer(Video video) =>
      Modular.to.pushNamed("videoplayer", arguments: video);
```
It's shouldn't be possible to use VideoPlayerPage in minimized mode.

-----
Regarding the limitation of `flutter_modular`:  
- It supports only material style transition and doesn't support essential for Flutter `PageRoute` custom transition which is more flexible. When I worked on VideoPlayer [I was forced to contribute to flutter_modular](https://github.com/minikin/modular/commit/9e4eb6734bd199868f97a4227ee314677164fce6) just to have the possibility to use basic Flutter features.

**Pull request merged**